.github/workflows/ci.yml

```diff
@@ -43,17 +51,63 @@ jobs:
           make deps
           make build
 
-  run_checks:
-    name: Run checks and tests
+  define-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      rust_toolchain_version: ${{ steps.rust.outputs.version }}
+      ocaml_version: ${{ steps.ocaml.outputs.version }}
+      node: ${{ steps.node.outputs.version }}
+      os: ${{ steps.os.outputs.version }}
+    steps:
+      - id: rust
+        run: echo 'version=["1.79", "1.80", "1.81", "1.82"]' >> "$GITHUB_OUTPUT"
+      # FIXME: currently not available for 5.0.0.
+      # It might be related to boxroot dependency, and we would need to bump
+      # up the ocaml-rs dependency
+      - id: ocaml
+        run: echo 'version=["4.14"]' >> "$GITHUB_OUTPUT"
+      - id: node
+        run: echo 'version=["20"]' >> "$GITHUB_OUTPUT"
+      - id: os
+        run: echo 'version=["ubuntu-latest", "ubuntu-24.04-arm"]' >> "$GITHUB_OUTPUT"
+
+  refresh-cache:
+    needs: ["define-matrix"]
+    strategy:
+      matrix:
+        rust_toolchain_version: ${{ fromJSON(needs.define-matrix.outputs.rust_toolchain_version) }}
+        os: ${{ fromJSON(needs.define-matrix.outputs.os) }}
+    runs-on: ${{ matrix.os }}
+    steps:
+
+      # if 'workflow_run'
+
+      - uses: actions/checkout@v4
+        if: ${{ github.event_name == 'workflow_run' }}
+      - uses: ./.github/actions/toolchain-shared
+        if: ${{ github.event_name == 'workflow_run' }}
+        with:
+          rust_toolchain_version: ${{ matrix.rust_toolchain_version }}
+      - name: start-fresh
+        if: ${{ github.event_name == 'workflow_run' }}
+        shell: bash
+        run: cargo clean
+
+      # else
+
+      - name: skip-refresh
+        if: ${{ github.event_name != 'workflow_run' }}
+        shell: bash
+        run: echo 'no need to refresh'
+
+  doc-and-spec:
```

Comment:
Nit: Do we want to run it for each Rust version? I do believe we did remove this. We should have a single job running the one that will be run on `master` to update the docs, and use a single Rust version. Can be done in a follow-up.

Reply:
See `./github/workflows/gh-page.yml`

Reply:
> I do believe we did remove this.

I'm wrong, sorry. We do run it for each run, you're right. I was confused by the diff.